lib/spack/spack/test/conftest.py

```diff
@@ -410,29 +416,30 @@ def database(tmpdir_factory, mock_packages, config):
     tmp_store = spack.store.Store(str(install_path))
     spack.store.store = tmp_store
 
-    _populate(tmp_store.db)
+    # If the cache exists just copy it in place, otherwise populate the
+    # store and create the cache.
+    if os.path.exists(str(_installation_cache.join('.spack-db'))):
+        _installation_cache.copy(install_path, mode=True, stat=True)
+    else:
+        _populate(tmp_store.db)
+        install_path.copy(_installation_cache, mode=True, stat=True)
 
     yield tmp_store.db
 
-    with tmp_store.db.write_transaction():
-        for spec in tmp_store.db.query():
-            if spec.package.installed:
-                PackageBase.uninstall_by_spec(spec, force=True)
-            else:
-                tmp_store.db.remove(spec)
```

Comment:
I like the fact that this code block actually tests `uninstall_by_spec()` at the end of testing.  Do we lose coverage by removing it?

It seems like to me that we should make `database` a session-scoped fixture, and `mutable_database` function scoped.  That would ensure that these uninstalls only happen once.

Reply:
I added that as a new test:

https://github.com/spack/spack/blob/2196ef6e9183f2ff6487caf2cdd87178eda2450b/lib/spack/spack/test/database.py#L652-L659